11_intro.md

```diff
@@ -0,0 +1,238 @@
+
+Scopul proiectului Clang este crearea unui concurent pentru compilatorul GCC. Dezvoltarea Clang se face in cadrul proiectului LLVM. Clang este un software liber, distribuit sub licenta BSD. In cadrul proiectului participa programatori de la Google, Apple, Microsoft si alte companii.
+
+### llvm
+
+LLVM (Low Level Virtual Machine) - proiect de infrastructura software pentru crearea compilatoarelor si a unor utilitati conexe. Proiectul consta dintr-un set de compilatoare pentru limbajele de programare de nivel inalt (__frontend__-uri), un sistem de optimizare, interpretare si compilare a codului in cod masina.
+
+LLVM, scris in C++, ofera optimizari de diferite niveluri, de la simpla eliminare a codului mort pana la optimizari complexe, cum ar fi vectorizarea, inlining-ul si multe altele. Din start in cadrul proiectului au fost realizate compilatoare pentru limbajele C si C++, dar in prezent LLVM suporta o gama larga de limbaje de programare, de exemplu: C#, Fortran, Haskell, Java (bytecode), JavaScript, Ruby, Rust, Scala, Swift si altele.
+
+LLVM poate crea cod masina pentru o gama larga de arhitecturi, inclusiv ARM, x86, x86-64, PowerPC, MIPS, SPARC, RISC-V si altele (inclusiv GPU de la Nvidia si AMD).
+
+Pentru compilarea codului C++ LLVM utilizeaza compilatorul GCC customizat sau CLang.
+
+## Medii de elaborare
+
+Un __mediu de elaborare__ (__IDE__ - Integrated Development Environment), altfel numit __mediu de programare__ sau __mediu de dezvoltare__, este o aplicatie care ofera un set de instrumente pentru dezvoltarea aplicatiilor software. Mediu de elaborare include de obicei un editor de text, un compilator, un depanator si alte instrumente necesare pentru dezvoltarea aplicatiilor software. In prezent, este dificil sa ne imaginam dezvoltarea aplicatiilor software fara utilizarea unui mediu de elaborare. Puteti scrie un mic program intr-un editor de text simplu si sa-l compilati folosind linia de comanda, dar in cazul dezvoltarii unor proiecte mari, acest lucru este neproductiv.
+
+Exista multe medii de elaborare pentru C++, o parte din ele sunt prezentate mai jos.
+
+### Code::Blocks
+
+![Code::Blocks](./images/11_1_codeblocks.png)
+
+[Code::Blocks](http://www.codeblocks.org) - un mediu de elaborare gratuit, destinat crearii aplicatiilor software, inclusiv C++. Code::Blocks este un mediu de elaborare multiplatforma, care permite utilizarea sa pe diferite sisteme de operare, inclusiv Windows, Linux si macOS. Code::Blocks este frecvent utilizat pentru a invata programarea in C++ in scoli si universitati.
+
+### Dev-C++
+
+![Dev-C++](./images/11_2_devcpp.png)
+
+[Dev-C++](https://sourceforge.net/projects/orwelldevcpp) este un mediu de elaborare gratuit pentru C++, destinat sistemului de operare Windows. Dev-C++ include un compilator MinGW, care permite compilarea programelor C++ sub Windows. Dev-C++ este un mediu de elaborare invechit, dar inca este utilizat pentru a invata programarea in C++ in scoli.

```

Comment:
Orwell Dev-C++ este învechit, dar [Embarcadero Dev-C++](https://github.com/Embarcadero/Dev-Cpp) este cel mai recent, suportând parțial C++20 (mai puțin învechit pentru curriculumul nostru?).